### PR TITLE
Upgrade to newest version of @daml/react

### PR DESCRIPTION
**Pull request by hurryabit**
This will remove the delay of seeing a new friend on your list after adding
them. I want to make sure the feature works as intended.

I've copied the files from the PR at
https://github.com/digital-asset/daml/pull/4368.

I've deliberately removed the tests as they don't get run here anyway.

---
This change is [Reviewable](https://reviewable.io/reviews/digital-asset/create-daml-app/12)

### CODE REVIEW

File: ui/src/daml-react/credentials.ts [hunk 1]
```diff
@@ -1,20 +1,19 @@
 import { decode } from 'jwt-simple';
 
-const LEDGER_ID: string = 'DAVL';
-
 export type Credentials = {
   party: string;
   token: string;
+  ledgerId: string;
```
Comment (rohanjr) [inline ui/src/daml-react/credentials.ts]: We'll also need to update the LoginScreen as well to use this `Credentials` type instead of the copy in `src/Credentials.ts`. I'll do that in another PR

File: ui/src/daml-react/DamlLedger.ts [hunk 2]
```diff
@@ -17,11 +17,7 @@ const DamlLedger: React.FC<Props> = (props) => {
     party: props.credentials.party,
     ledger: new Ledger(props.credentials.token),
   }), [props.credentials, store, dispatch])
-  return (
-    <DamlLedgerContext.Provider value={state}>
-      {props.children}
-    </DamlLedgerContext.Provider>
-  );
+  return React.createElement(DamlLedgerContext.Provider, {value: state}, props.children);
```
Comment (rohanjr) [inline ui/src/daml-react/DamlLedger.ts]: What's the reason for this change here? Is it the desire to move away from tsx/jsx generally?

File: ui/src/daml-react/credentials.ts (hunk 1 above)
Comment (hurryabit) [reply ui/src/daml-react/credentials.ts]: Thanks.

File: ui/src/daml-react/DamlLedger.ts (hunk 2 above)
Comment (hurryabit) [reply ui/src/daml-react/DamlLedger.ts]: See https://github.com/digital-asset/daml/pull/4273 for an explanation.
